frame/contracts/src/lib.rs

```diff
@@ -361,20 +361,20 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			#[pallet::compact] endowment: BalanceOf<T>,
 			#[pallet::compact] gas_limit: Weight,
-			code: Vec<u8>,
-			data: Vec<u8>,
-			salt: Vec<u8>,
+			code: Box<Vec<u8>>,
+			data: Box<Vec<u8>>,
+			salt: Box<Vec<u8>>,
```

Comment:
Why do we box `Vec`? It already allocated its elements on the heap. There are more places where this happened. I just mark this one as example.

Reply:
I was surprised too but a vec is 24 bytes https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=076c59fc288d9fe6bb17e1c7c92e35a7

so this extrinsic variant fields are: `16 + 8 + 24 * 3` added to the variant indices `1 for pallet variant, 1 for call variant` it is 98 bytes

Reply:
Ohh. I expected it to be much smaller. Then go ahead.

Reply:
it is heap, but it is more with `Vec` https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9f93db0b5c42a9ece4f8658a34d821d0